### PR TITLE
feat(hashrego): venue-weekend campout heuristic — closes #1560

### LINKED ISSUE #1560: Multi-day weekend events: surface child trails under a series parent + support date-range campouts
## Summary

HashTracks has schema + merge support for multi-day **series** (`Event.parentEventId`, `Event.isSeriesParent`, `RawEventData.seriesId`, [`linkMultiDaySeries`](https://github.com/johnrclem/hashtracks-web/blob/main/src/pipeline/merge.ts#L1638)) but the UX layer hides children everywhere and the schema can't represent single-registration multi-day events at all. Three real-world weekend events expose three distinct patterns that we need to model end-to-end.

Filed as the follow-up to #1421 (which dropped duplicate SFH3 umbrella VEVENTs at the adapter level rather than modeling them as a parent series — the right narrow fix, but it sidestepped the broader UX gap).

## Background — what exists today

- **Schema is ready (mostly)**: [`prisma/schema.prisma:328-405`](https://github.com/johnrclem/hashtracks-web/blob/main/prisma/schema.prisma#L328) has `Event.parentEventId`, `Event.isSeriesParent`, `childEvents` relation, index on `parentEventId`.
- **Merge is ready**: [`linkMultiDaySeries`](https://github.com/johnrclem/hashtracks-web/blob/main/src/pipeline/merge.ts#L1638) accumulates seriesId groups and marks the earliest as parent, rest as children. Best-effort.
- **Adapter contract is ready**: [`RawEventData.seriesId`](https://github.com/johnrclem/hashtracks-web/blob/main/src/adapters/types.ts#L41) is plumbed.
- **Only Hash Rego emits seriesId** today, and only for the `MM/DD HH:MM PM to MM/DD HH:MM PM` date-range pattern ([`parser.ts:399-423`](https://github.com/johnrclem/hashtracks-web/blob/main/src/adapters/hashrego/parser.ts#L399)).
- **UI hides children**: consumer queries in [`page.tsx:29`](https://github.com/johnrclem/hashtracks-web/blob/main/src/app/page.tsx#L29), [`kennels/[slug]/page.tsx:101`](https://github.com/johnrclem/hashtracks-web/blob/main/src/app/kennels/[slug]/page.tsx#L101) (and other surfaces) filter `parentEventId: null`. Hash Rego's existing Day 2 / Day 3 weekend rows are already invisible today.
- **Schema gap**: `Event.date` is a single `DateTime`. No `endDate`. There's no way to model "weekend at one venue with one registration" without faking up child rows.

### Example 1 — SFH3 Bay 2 Blackout 2026 (umbrella + per-day trails, iCal)
URL: https://www.sfh3.com/events/134

The iCal feed publishes one umbrella + four trails:

| Type | URL | Date | Title |
|------|-----|------|-------|
| umbrella | `/events/134` | 2026-05-14 (all-day) | Bay 2 Blackout 2026 |
| trail | `/runs/6485` | 2026-05-15 18:00 PDT | SFH3: Friday Turkey / Eagle Run & Pub Crawl |
| trail | `/runs/6490` | 2026-05-16 11:30 PDT | SFH3: Saturday Ballbuster |
| trail | `/runs/6487` | 2026-05-16 13:30 PDT | SFH3: Saturday Turkey / Eagle Run & Pub Crawl |
| trail | `/runs/6488` | 2026-05-17 08:00 PDT | Sunday B2B Run / Walk |

The umbrella detail page explicitly links to each child trail (`B2B Day 1/2/3` hrefs). The umbrella carries weekend-level context (themes, fees, registration tiers, lodging info) that no individual trail has.

**Desired UX**: "Bay 2 Blackout 2026" appears as a single weekend card. Card shows day-of-week pills (Fri/Sat/Sun) backed by the four trail child events. Tapping a pill expands the per-day trail (time, hares, location). Direct visit to a trail (`/hareline/<runs-6485-id>`) still works and shows a back-link to the umbrella.

**Current PR (#1531) does NOT model this** — it drops the umbrella at the adapter level when a same-day trail exists. The umbrella's description is lost in the collision case. This is fine as a stopgap; the proper fix is series modeling.

### Example 2 — NYC H3 5-Boro Pub Crawl 2026 (multi-leg weekend, cross-kennel)
URL: https://hashrego.com/events/nych3-5-boro-pub-crawl-2026

`og:description` excerpt:
```
NYCH3 5-BORO PUB CRAWL 2026
A NYCH3 Hash Weekend — June 26–28, 2026

FRIDAY 6/26 — GGFM Strawberry Moon Trail
Greater Gotham Full Moon Hash #435 — Show 7:15 PM | Go 7:30 PM

SATURDAY 6/27 — 12th Annual 5-Borough Pub Crawl
12:30 PM | Start location details incuming!

SUNDAY 6/28 — NYC Pride March Watch Party!
Time TBA | Greenwich Village
```

The current Hash Rego parser only recognizes the `MM/DD HH:MM PM to MM/DD HH:MM PM` range pattern. The 5-Boro description has three separate dated section headers without a single `to`-style range, so `parseDateRangeFromDescription` returns null. The adapter falls back to the single index date (Saturday 06/27) and **silently drops Friday + Sunday entirely**.

Two things to fix:
1. **Parser**: extend Hash Rego date detection to recognize `**DAY M/D —**` section headers and emit one RawEventData per dated leg with shared `seriesId`.
2. **UI**: same multi-day rendering work as Example 1.

**Bonus complication**: the Friday leg is hosted by **GGFM** (Greater Gotham Full Moon Hash), not NYCH3. The series parent is NYCH3 but the Friday child belongs to GGFM for attendance purposes. `EventKennel.isPrimary` covers per-event kennel attribution already, but the series-linker doesn't reason about cross-kennel children yet.

### Example 3 — Madison H3 Token Run Campout 2026 (single-registration venue weekend)
URL: https://hashrego.com/events/madisonh3-token-run-campout-2026

`og:title`: `05/29 MadisonH3 Token Run Campout 2026`

`og:description` excerpt:
```
Cum to Token Creek on the outskirts of Madison for a weekend of camping...

Friday:
Arrive, set up your tent, get a drink
Pittsburgh Beer Fest!
FULL MOON late night woodland dash trail...

Saturday:
Olympdic Games... Epic Saturday Trail... Late Night Stumble-a-thon...

Sunday:
Hair of the Dog and GTFO! Pack up by 12:30...
```

This is **qualitatively different** from Examples 1 and 2:
- ONE registration, ONE venue (Token Creek campground), ONE Venmo payment
- Per-day labels are *informal programming notes*, not separate registration/trail rows
- A hasher who attends gets ONE logbook entry, not three
- The `og:description` uses day-of-week labels (`Friday:`, `Saturday:`, `Sunday:`) with NO explicit dates per section, so the current parser regex won't match

**Current behavior**: emits a single Event on 05/29 (the index startDate). The fact that it spans Friday–Sunday is lost. UI shows it as a one-day event.

**Desired UX**:
- Single event card "MadisonH3 Token Run Campout 2026" displayed with a **date range** ("May 29–31, 2026") instead of a single date
- Per-day programming surfaces as a description section, not as child events
- No spawned child rows in the DB

**Implication**: this needs more than series modeling. It needs either:
- A new `Event.endDate` schema field (cleanest), OR
- A convention where the parent event has duration metadata in description/EventLinks (hackier)

Plus a parser change: detect "campout"/"weekend" in title/description + day-of-week labels → derive a date range from the index startDate + a sensible duration heuristic.

## Three patterns side-by-side

| Pattern | Source | Has children? | Date model |
|---------|--------|---------------|------------|
| Umbrella + per-day trails | SFH3 iCal | Yes (4) | Parent date + 4 child dates |
| Multi-leg weekend | Hash Rego HTML | Yes (3, cross-kennel) | Parent date + 3 child dates |
| Single-reg venue weekend | Hash Rego HTML | No | Date range (start + end) |

Solving all three gives us a template for nearly every weekend event in the network.

## Proposed scope (suggest splitting into sub-tasks)

1. **Schema**: add `Event.endDate: DateTime?` (covers Example 3; non-breaking).
2. **UI — series**: surface parent events with child-count pills + expandable per-day rows on the hareline + kennel page. Remove `parentEventId: null` filter from the queries that should display series, keep it on the queries that shouldn't (calendar grid).
3. **UI — date range**: render `endDate` when present (date-range display on cards; "May 29–31, 2026").
4. **Adapter — SFH3 iCal**: detect umbrella+trail clusters, emit shared `seriesId`. Replace the current "drop umbrella" suppression with proper series linking. Salvage umbrella description into the parent Event's description.
5. **Adapter — Hash Rego**: extend `parseDateRangeFromDescription` to recognize per-day section headers (`**FRIDAY M/D —**`). Emit one RawEventData per dated leg with shared seriesId, handling cross-kennel children via `kennelTags`.
6. **Adapter — Hash Rego**: detect "campout/weekend" pattern with day-of-week labels (no explicit per-day dates). Emit a single Event with `endDate` set from a heuristic (e.g. startDate + 2 days for Fri→Sun).

## Critical files

- [`src/pipeline/merge.ts:1638-1665`](https://github.com/johnrclem/hashtracks-web/blob/main/src/pipeline/merge.ts#L1638) — `linkMultiDaySeries`
- [`src/adapters/types.ts:41`](https://github.com/johnrclem/hashtracks-web/blob/main/src/adapters/types.ts#L41) — `RawEventData.seriesId`
- [`prisma/schema.prisma:328-405`](https://github.com/johnrclem/hashtracks-web/blob/main/prisma/schema.prisma#L328) — `Event.parentEventId`, `Event.isSeriesParent`, `childEvents`
- [`src/adapters/hashrego/parser.ts:264-309`](https://github.com/johnrclem/hashtracks-web/blob/main/src/adapters/hashrego/parser.ts#L264) — multi-day splitter
- [`src/adapters/hashrego/parser.ts:399-423`](https://github.com/johnrclem/hashtracks-web/blob/main/src/adapters/hashrego/parser.ts#L399) — `parseDateRangeFromDescription` (gaps for Examples 2 + 3)
- [`src/app/page.tsx:29`](https://github.com/johnrclem/hashtracks-web/blob/main/src/app/page.tsx#L29), [`src/app/kennels/[slug]/page.tsx:101`](https://github.com/johnrclem/hashtracks-web/blob/main/src/app/kennels/[slug]/page.tsx#L101) — UI consumer queries that hide children
- [`src/adapters/html-scraper/sfh3-detail-enrichment.ts`](https://github.com/johnrclem/hashtracks-web/blob/main/src/adapters/html-scraper/sfh3-detail-enrichment.ts) — current `suppressSFH3UmbrellaDuplicates` to replace with series linking

### PR DESCRIPTION
**Pull request by johnrclem**
## Summary

Final PR in the multi-day series stack ([PR A #1627](https://github.com/johnrclem/hashtracks-web/pull/1627) → [PR B #1630](https://github.com/johnrclem/hashtracks-web/pull/1630) → this). Adds a third strategy to Hash Rego date extraction for events that register as ONE row but span multiple days — MadisonH3 Token Run Campout was the motivating case. The description mentions weekday names (Friday/Saturday/Sunday) with no explicit per-day dates AND a "weekend" / "campout" / "retreat" / "rendezvous" trigger word; the heuristic infers the inclusive last day by counting forward from the index entry's start date.

- New `detectVenueWeekendEndDate(description, title, startDateStr)` exported helper.
- `ParsedEvent.endDate?: string` new optional field.
- `extractDates` now threads `title` and surfaces `endDate` when the heuristic fires on the single-day fallback path.
- `splitToRawEvents` single-day branch emits `endDate` when present (no `seriesId`, no `seriesParent`, no children — one row, one registration, just a date range).

## How it differs from PR B's `parseDayHeaderSections`

| | PR B | PR C |
|---|---|---|
| Trigger | `**DAY N M/D —**` per-day section headers in description | "campout"/"weekend"/"retreat"/"rendezvous" + ≥2 weekday mentions |
| Output | Multi-day series with N children + explicit parent | Single row with `endDate` set |
| UI treatment | Date-range chip + `+ N trails` badge + expandable timeline | Date-range chip + `Weekend` badge, no expansion |

UI behavior comes from PR A's `hasDateRange` check (events with `endDate != null` AND `isSeriesParent === false` render the Madison-style date-range pill without an expandable timeline).

## Heuristic safety
- **Trigger gating**: Without the keyword (`camp(out)|weekend|retreat|rendezvous`), the weekday scan never runs. A single-day trail whose description happens to name "Saturday brunch on-after" doesn't spuriously expand.
- **`\b`-anchored weekday match**: "Sat" inside "Saturn" / "Satellite" doesn't fire.
- **Forward-counting offset**: `(dow - startDow + 7) % 7` keeps the inferred range inside a single 7-day window — the longest a reasonable hashing weekend would be.
- **Single-day collapse**: If every mentioned weekday resolves to the start day, return `null` (no spurious 0-day "range").

## Tests
- 7 `it.each` cases covering the motivating Madison case, title-only triggers, "camp out" with space, abbreviated weekday names, `rendezvous` trigger, Saturday-start single-wrap, and a NYE year-boundary case (Fri 2027-12-31 → Sun 2028-01-02).
- 4 negative cases: no trigger word, single weekday only, all mentions collapse to start day, "Saturn" doesn't match "Sat".
- Full integration test using a synthetic Madison Token Run HTML fixture (1 row, `endDate` present, no `seriesId`/`seriesParent`).
- Regression guard: single-day events without weekend triggers keep `endDate` undefined.

## Verification
- [x] `npx tsc --noEmit` — clean
- [x] `npm run lint` — 0 errors, 21 pre-existing warnings
- [x] `npm test` — 7199 tests pass (+13 new for venue-weekend)
- [ ] Post-merge: Hash Rego re-scrape on prod will surface MadisonH3-style venue weekends as date-range cards in the hareline.

Closes #1560.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by johnrclem** [comment]
@claude @codex review

File: src/adapters/hashrego/parser.ts [hunk 1]
```diff
@@ -638,22 +652,125 @@ function parseDateRangeFromDescription(
...
+  // DST boundary (matches the project's UTC-noon date convention from
+  // CLAUDE.md §F.4).
+  const startDate = new Date(`${startDateStr}T12:00:00Z`);
+  if (Number.isNaN(startDate.getTime())) return null;
+  const startDow = startDate.getUTCDay();
+
+  let maxOffset = 0;
+  for (const dow of mentioned) {
+    const offset = (dow - startDow + 7) % 7;
+    if (offset > maxOffset) maxOffset = offset;
```
[the remark this answers is not in the document]
Comment (johnrclem) [reply src/adapters/hashrego/parser.ts:749]: Fixed in 988ef172 — capped forward-offset at MAX_FORWARD_OFFSET=4. A Fri-start campout mentioning Thursday now ignores the Thursday as a backward (prelube) reference; max plausible weekend length is Thu→Mon (4 days), which still resolves correctly. Added the exact Codex repro as a regression test + an inverse 4-day Thu→Mon test.

File: src/adapters/hashrego/parser.ts [hunk 2]
```diff
@@ -638,22 +652,125 @@ function parseDateRangeFromDescription(
...
+  }
+  const mentioned = new Set<number>();
+  for (const m of description.matchAll(WEEKDAY_NAME_RE)) {
+    const lc = m[1].toLowerCase();
+    if (!WEEKDAY_NAMES_SET.has(lc)) continue;
+    const prefix = lc.slice(0, 3);
+    const dow = WEEKDAY_PREFIXES.indexOf(prefix);
+    if (dow >= 0) mentioned.add(dow);
+  }
+  if (mentioned.size < 2) return null;
```
[the remark this answers is not in the document]
Comment (johnrclem) [reply src/adapters/hashrego/parser.ts:726]: Addressed in 988ef172 — replaced the manual getUTC*/padStart chain with setUTCDate + toISOString().split('T')[0]. Didn't extract to a shared DateUtils because this is a one-shot end-of-range computation, not a repeated pattern; happy to add one if you want centralization across multiple parsers.
